app/controllers/institutions_controller.rb

```diff
@@ -112,30 +100,21 @@ def no_data_allowed
     redirect_to home if home != no_data_allowed_institutions_path
   end
 
+  # NOTE: kept for backward compatibility, delete this route/action after next release
   def new_from_invite_data
-    pending_invite = PendingInstitutionInvite.find_by(
-      invited_user_email: current_user.email,
-      id: params["pending_institution_invite_id"].presence,
-      status: 'pending'
-    )
-    unless pending_invite.nil?
-      @institution = current_user.institutions.new
-      @institution.kind = pending_invite.institution_kind
-      @institution.name = pending_invite.institution_name
-      @institution.pending_institution_invite = pending_invite
-      render action: 'new'
-    else
-      session[:user_return_to] = nil
-      no_data_allowed
-    end
+    new
+    render :new
   end
 
   private
 
-  def accept_pending_invite
-    if @pending_invite && current_user.email == @pending_invite.invited_user_email
-      @pending_invite.status = 'accepted'
-      @pending_invite.save!
+  def load_pending_invite!
+    return unless id = params[:institution] ? params[:institution][:pending_institution_invite_id] : params[:pending_institution_invite_id]
```

Comment:
There's so much in this line... what about this instead:
```suggestion
    id = params.dig(:institution, :pending_institution_invite_id) || params[:pending_institution_invite_id] || return
```

Reply:
I thought of `#dig` but we need Ruby 2.3, or the patch from the rails5 upgrade, or I can just use the custom `#get_in` method we currently have (then replace for `#dig` when rebasing the rails5 branch).